dao/src/main/kotlin/Database.kt

```diff
@@ -210,4 +210,20 @@ fun <T : LongEntity> LongEntityClass<T>.findSingle(condition: ConditionBuilder):
  * makes sure that the result does not contain any duplicates.
  */
 fun <M, D> mapAndDeduplicate(modelObjects: Collection<M>?, mapper: (M) -> D): SizedIterable<D> =
-    SizedCollection(modelObjects?.mapTo(mutableSetOf(), mapper).orEmpty())
+    SizedCollection(mapToSet(modelObjects, mapper).orEmpty())
+
+/**
+ * Map the given [modelObjects] collection to a [Set] using the given [mapper] function and check whether the result
+ * equals to the given [Collection] of [dataObjects]. This is used by _find_ functions in some Dao implementations to
+ * deal with collections of child elements that cannot have duplicates and for which order does not matter.
+ */
+fun <M, D> mapAndCompare(modelObjects: SizedIterable<M>, dataObjects: Collection<D>, mapper: (M) -> D): Boolean {
+    val mapped = mapToSet(modelObjects, mapper).orEmpty()
```

Comment:
I wonder if we could not:

1. Convert the `SizedIterable` to set without mapping
2. Compare the sizes and fail early
3. Map the set obtained in 1 using `mapper.`
4. Do the `compareAll` comparison

This looks like a micro optimization, but this is something done quite a lot in the ORT Server backend, so maybe it would be worthwhile.